src/Resque/Component/Worker/Factory/WorkerFactory.php

```diff
@@ -28,45 +29,47 @@ class WorkerFactory implements WorkerFactoryInterface
     protected $eventDispatcher;
 
     /**
+     * @var SystemInterface
+     */
+    protected $system;
+
+    /**
      * Constructor
      *
      * @param QueueFactoryInterface $queueFactory
      * @param JobInstanceFactoryInterface $jobFactory
      * @param EventDispatcherInterface $eventDispatcher
+     * @param SystemInterface $system
      */
     public function __construct(
         QueueFactoryInterface $queueFactory,
         JobInstanceFactoryInterface $jobFactory,
-        EventDispatcherInterface $eventDispatcher
+        EventDispatcherInterface $eventDispatcher,
+        SystemInterface $system
     ) {
         $this->queueFactory = $queueFactory;
         $this->jobInstanceFactory = $jobFactory;
         $this->eventDispatcher = $eventDispatcher;
-    }
-
-    // @todo work out how to only have this in one place (like environment->getHostname), or something like that.
-    public function getHostname(){
-        if (function_exists('gethostname')) {
-            $hostname = gethostname();
-        } else {
-            $hostname = php_uname('n');
-        }
-        return $hostname;
+        $this->system = $system;
     }
 
     /**
      * Create worker
      *
+     * @param Process $process
      * @return WorkerInterface
      */
-    public function createWorker()
+    public function createWorker(Process $process = null)
     {
         $worker = new Worker(
             $this->jobInstanceFactory,
             $this->eventDispatcher
         );
 
-        $worker->setHostname($this->getHostname());
+        $worker->setHostname($this->system->getHostname());
+        if($process) {
+            $worker->setProcess($process);
```

Comment:
Hey dude.

From my point of view adding `Process $process = null` is a breach of the WorkerFactoryInterface::createWorker() signature breaking E_STRICT, and I'm not sure what value it it adds when WorkerInterface, has setProcess() so you are a guaranteed to have it on the returned worker.

I am curious as to why you added it, in case there is a problem this is solving for you.

As an FYI, I am going to accept the request as the rest is useful, but I will be stripping this particular change from it as it isn't quite right as it is.